datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -98,6 +104,13 @@
   private final DynamicMetricsManager _dynamicMetricsManager;
   private final String _metricsNamesPrefix;
 
+  // A lock used to synchronized access to operations performed on the _kafkaProducer object
+  private final Object _producerLock = new Object();
+
+  // An executor to spawn threads to close the producer.
+  private final ExecutorService _producerCloseExecutorService = Executors.newSingleThreadExecutor(
+      new ThreadFactoryBuilder().setNameFormat("KafkaProducerWrapperClose-%d").build());
```

Comment:
I am guessing we can now use setName without the %d with singleThreadExecutor?

Reply:
as discussed, there is no `setName` API for the  ThreadFactoryBuilder,  so leaving this as is.